addons/website_blog/controllers/main.py

```diff
@@ -55,7 +55,22 @@ def nav_list(self, blog=None):
 
         return OrderedDict((year, [m for m in months]) for year, months in itertools.groupby(groups, lambda g: g['year']))
 
-    def _prepare_blog_values(self, blogs, blog=False, date_begin=False, date_end=False, tags=False, state=False, page=False, search=None):
+    def _get_blog_post_search_options(self, blog=None, active_tags=None, date_begin=None, date_end=None, state=None, **post):
+        return {
+            'displayDescription': True,
+            'displayDetail': False,
+            'displayExtraDetail': False,
+            'displayExtraLink': False,
+            'displayImage': False,
+            'allowFuzzy': not post.get('noFuzzy'),
```

Comment:
Why is this replacing `'allowFuzzy': not request.params.get('noFuzzy'),` ?

Reply:
This is a facility that was introduced initially to be able to disable the fuzzy search in case of problems (and indeed problems were found).
`allowFuzzy: False` makes it possible to disable the fuzzy search.
The `s_searchbar` can be setup with a `data-no-fuzzy` that has two impacts:
- add a `allowFuzzy: false` to the autocomplete search options
- add a `noFuzzy` hidden input inside the form
```js
        if (this.$input.data('noFuzzy')) {
            $("<input type='hidden' name='noFuzzy' value='true'/>").appendTo(this.$input);
        }
```
Therefore, for the search routes, the value must be extracted from the `noFuzzy` parameter.

The reason behind the negative form is mostly because we wanted a default behavior of "with fuzzy" when nothing was specified. Not a very valid reason, but that's how it ended up.
We could rename stuff and swap things around - but then we'll also need upgrade scripts about this.

Reply:
I meant, why do you change this line in this PR and thus have to add **kw to both methods (the already existing one and the new one)

Reply:
Oh oops !
Because my understanding is that the purpose of extracting those methods that obtain these options is about being able to override them to possibly extract further parameters added by customizations - and it therefore will help to have access to additional request parameters.
But indeed, I could instead keep relying on `request.params` - and adapt the methods in the other modules to do it similarly as well.
Whichever way, I'd rather have those similar methods proceed in the same way - no other justification than personal taste (less differences ~ less to remember). If we switch them all to `request.params`, I would also prefer to adapt the existing one in `website_sale`.